samples/bluetooth/central_otc/src/main.c

```diff
@@ -568,6 +590,10 @@ static int on_obj_data_read(struct bt_ots_client *ots_inst, struct bt_conn *conn
 		printk("Object total received %d\n", len + offset);
 		print_hex_number(obj_data_buf, len + offset);
 		(void)memset(obj_data_buf, 0, OBJ_MAX_SIZE);
+		otc_checksum_work.offset = 0;
+		otc_checksum_work.len = otc.cur_object.size.cur;
+		k_work_schedule(&otc_checksum_work.work, K_MSEC(50));
+
```

Comment:
Why do we need a workqueue item for this? At this point we are just sending a GATT request in the `otc_checksum_work_fn` function, and that should be fine to do in this callback directly, right?

Reply:
OTS client API couldn't be called in OTS client cb function.
obj_data_read cb is called [here](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/services/ots/ots_client.c#L194) before [cur_inst released](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/bluetooth/services/ots/ots_client.c#L209).
Which makes oacp_checksum return -EBUSY [here](https://github.com/zephyrproject-rtos/zephyr/pull/50451/files#diff-7278e3a1c0d69370760b526260965a6fd17f04254467a377f08592d1c11ccb6aR1302) .

Reply:
Fair point. Is that mentioned in the API documentation anywhere?

Reply:
It doesn't in API documentation.